src/posthog-surveys.ts

```diff
@@ -386,15 +386,19 @@ export class PostHogSurveys {
         return assignableWindow.__PosthogExtensions__.canActivateRepeatedly(survey)
     }
 
-    canRenderSurvey(surveyId: string) {
+    canRenderSurvey(surveyId: string): SurveyRenderReason | null {
         if (isNullish(this._surveyManager)) {
             logger.warn('init was not called')
-            return
+            return { visible: false }
         }
+        let renderReason: SurveyRenderReason | null = null
         this.getSurveys((surveys) => {
             const survey = surveys.filter((x) => x.id === surveyId)[0]
-            this._surveyManager.canRenderSurvey(survey)
+            if (survey) {
+                renderReason = { ...this._surveyManager.canRenderSurvey(survey) }
+            }
```

Comment:
I think we need an `else` branch here with `visible: false` and `disabledReason: 'Survey not found'`

Reply:
agreed, addressed both comments